src/app/utils/hooks/useIsDeepLink.ts

```diff
@@ -0,0 +1,55 @@
+import { useIsFocused } from "@react-navigation/native"
+import { matchRoute } from "app/routes"
+import { useEffect, useState } from "react"
+import { Linking } from "react-native"
+
+/**
+ * This is a hook that returns whether or not the user came from a deep link
+ * (defined as a direct navigation to a route other than "/").
+ *
+ * This can be used to avoid rendering content in previous screens in react-navigation history
+ *
+ * @returns {isDeepLink: boolean | null} isDeepLink is true if the user came from a deep link
```

Comment:
I'd add something like this to the description to make sure the hook is used correctly. (`null` vs `false` value).

```suggestion
 * @returns {isDeepLink: boolean | null}` isDeepLink` is true if the user came from a deep link. Initially, it is set to `null` while retrieving the deep link URL asynchronously and will be set to `false` if retrieving the `URL` fails.
```